src/app/(pages)/mobility.tsx

```diff
@@ -61,60 +117,123 @@ export default function Mobility(): JSX.Element {
                 setLoadingState(LoadingState.ERROR)
             })
     }
-
     return (
         <ScrollView
             refreshControl={
-                loadingState !== LoadingState.LOADING ? (
+                loadingState !== LoadingState.LOADING &&
+                loadingState !== LoadingState.LOADED ? (
                     <RefreshControl
                         refreshing={loadingState === LoadingState.REFRESHING}
-                        onRefresh={onRefresh}
+                        onRefresh={() => {
+                            onRefresh(mobilityKind)
+                        }}
                     />
                 ) : undefined
             }
         >
-            {loadingState === LoadingState.LOADING && (
-                <View style={styles.loadingContainer}>
-                    <ActivityIndicator size="small" color={colors.primary} />
-                </View>
-            )}
-            {loadingState === LoadingState.ERROR && (
-                <View>
-                    <Text style={[styles.errorMessage, { color: colors.text }]}>
-                        {error?.message}
-                    </Text>
-                    <Text style={[styles.errorInfo, { color: colors.text }]}>
-                        An error occurred while loading the data.{'\n'}Pull down
-                        to refresh.
-                    </Text>
+            <ToggleRow
+                items={displayTypes}
+                selectedElement={mobilityKind}
```

Comment:
When restarting the app, no transport mode is preselected, neither the previously saved one, nor a default selection.